x-pack/plugins/apm/common/service_groups.ts

```diff
@@ -20,3 +32,47 @@ export interface SavedServiceGroup extends ServiceGroup {
   id: string;
   updatedAt: number;
 }
+
+export const SERVICE_GROUP_SUPPORTED_FIELDS = [
+  AGENT_NAME,
+  SERVICE_NAME,
+  SERVICE_ENVIRONMENT,
+  SERVICE_LANGUAGE_NAME,
+  LABELS,
+];
+
+export function isSupportedField(fieldName: string) {
+  return (
+    fieldName.startsWith(LABELS) ||
+    SERVICE_GROUP_SUPPORTED_FIELDS.includes(fieldName)
+  );
+}
+
+export function validateServiceGroupKuery(kuery: string) {
+  try {
+    const kueryFields = getKueryFields([fromKueryExpression(kuery)]);
+    const unsupportedKueryFields = kueryFields.filter(
+      (fieldName) => !isSupportedField(fieldName)
+    );
+    if (unsupportedKueryFields.length > 0) {
+      return {
+        isValid: false,
+        isParsingError: false,
+        message: i18n.translate(
+          'xpack.apm.serviceGroups.invalidFields.message',
+          {
+            defaultMessage:
+              'Query filter for service group does not support fields [{unsupportedFieldNames}]',
+            values: {
+              unsupportedFieldNames: unsupportedKueryFields.join(', '),
+            },
+          }
+        ),
+      };
+    } else {
+      return { isValid: true, isParsingError: false, message: '' };
+    }
+  } catch (error) {
+    return { isValid: false, isParsingError: true, message: error.message };
+  }
+}
```

Comment:
`isValidFields` and `isValidSyntax` is a bit more clear. Also, flipping the conditional `unsupportedKueryFields.length > 0`) some indention can be removed:

```suggestion
export function validateServiceGroupKuery(kuery: string) {
  try {
    const kueryFields = getKueryFields([fromKueryExpression(kuery)]);
    const unsupportedKueryFields = kueryFields.filter(
      (fieldName) => !isSupportedField(fieldName)
    );
    if (unsupportedKueryFields.length === 0) {
      return { isValidFields: true, isValidSyntax: true };
    }

    return {
      isValidFields: false,
      isValidSyntax: true,
      message: i18n.translate('xpack.apm.serviceGroups.invalidFields.message', {
        defaultMessage:
          'Query filter for service group does not support fields [{unsupportedFieldNames}]',
        values: {
          unsupportedFieldNames: unsupportedKueryFields.join(', '),
        },
      }),
    };
  } catch (error) {
    return {
      isValidFields: false,
      isValidSyntax: false,
      message: error.message,
    };
  }
}
```